idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt

```diff
@@ -47,24 +48,25 @@ class RedundantCompanionReferenceInspection : AbstractKotlinInspection() {
             when (selectorDescriptor) {
                 is PropertyDescriptor -> {
                     val name = selectorDescriptor.name
-                    if (containingClass.findPropertyByName(name.asString()) != null) return
+                    if (containingClassDescriptor.findMemberVariable(name.asString()) != null) return
                     val variable = expression.getResolutionScope().findVariable(name, NoLookupLocation.FROM_IDE)
                     if (variable != null && variable.isLocalOrExtension(containingClassDescriptor)) return
                 }
                 is FunctionDescriptor -> {
                     val name = selectorDescriptor.name
-                    val function = containingClass.findFunctionByName(name.asString())?.descriptor
-                        ?: expression.getResolutionScope().findFunction(name, NoLookupLocation.FROM_IDE)?.takeIf {
+                    val functions = containingClassDescriptor.collectMemberFunction(name.asString()) + listOfNotNull(
+                        expression.getResolutionScope().findFunction(name, NoLookupLocation.FROM_IDE)?.takeIf {
                             it.isLocalOrExtension(containingClassDescriptor)
                         }
-                    if (function is FunctionDescriptor) {
-                        val functionParams = function.valueParameters
-                        val calleeParams =
-                            (selectorExpression as? KtCallExpression)?.calleeExpression?.getCallableDescriptor()?.valueParameters.orEmpty()
-                        if (functionParams.size == calleeParams.size &&
-                            functionParams.zip(calleeParams).all { it.first.type == it.second.type }
-                        ) return
-                    }
+                    )
+                    if (functions.any {
+                            val functionParams = it.valueParameters
+                            val calleeParams = (selectorExpression as? KtCallExpression)?.calleeExpression?.getCallableDescriptor()
+                                ?.valueParameters.orEmpty()
+                            functionParams.size == calleeParams.size && functionParams.zip(calleeParams).all { param ->
```

Comment:
Here we can have problems with default parameters. I think it's better to simplify this part and check just by name.